odoo/addons/test_lint/tests/_odoo_checker_gettext.py

```diff
@@ -22,15 +22,23 @@ class OdooBaseChecker(BaseChecker):
             'Bad usage of _, _lt function.',
             'gettext-variable',
             'See https://www.odoo.com/documentation/master/developer/misc/i18n/translations.html#variables'
-        )
+        ),
+        'E8505': (
+            'Usage of _, _lt function with multiple unnamed placeholders',
+            'gettext-placeholders',
+            'Use keyword arguments when you have multiple placeholders',
+        ),
     }
 
-    @only_required_for_messages('gettext-variable')
+    @only_required_for_messages('gettext-variable', 'gettext-placeholders')
     def visit_call(self, node):
         if isinstance(node.func, astroid.Name) and node.func.name in ('_', '_lt'):
             first_arg = node.args[0]
-            if not (isinstance(first_arg, astroid.Const) and isinstance(first_arg.value, str)):
-                self.add_message('gettext-variable', node=node)
+            if isinstance(first_arg.value, str):
+                if not isinstance(first_arg, astroid.Const):
+                    self.add_message('gettext-variable', node=node)
+                elif str(first_arg.value).count('%s') >= 2:
+                    self.add_message('gettext-placeholders', node=node)
```

Comment:
what's about the other `%` placeholders like `%d`?

```suggestion
                # move those 3 statements on top of the file
                import re
                # https://docs.python.org/2.6/library/stdtypes.html#string-formatting-operations
                PLACEHOLDER_REGEXP = re.compile(r"""
                    %
                    [#0\- +]*          # convertion flag
                    (?:\d+|\*)?        # minimum field width
                    (?:\.(?:\d+|\*))?  # precision
                    [hlL]?             # length modifier
                    [bcdeEfFgGnosxX]   # convertion type
                """, re.VERBOSE)  # strip all whitespaces

                if not isinstance(first_arg, astroid.Const):
                    self.add_message('gettext-variable', node=node)
                elif len(PLACEHOLDER_REGEXP.findall(str(first_arg.value))) >= 2:
                    self.add_message('gettext-placeholders', node=node)
```
[re.VERBOSE](https://docs.python.org/3/library/re.html#re.VERBOSE)

Result for common placeholders
```py
>>> PLACEHOLDER_REGEXP.findall("hello %s %d %05d %.5f %% world")
['%s', '%d', '%05d', '%.5f']
```

Reply:
is the `str()` conversion inside `str(first_arg.value)` necessary?

Reply:
We were just about to implement something like that. Thanks for writing the code for us 😄

Reply:
> is the `str()` conversion inside `str(first_arg.value)` necessary?
@Julien00859 

`_(2)` → `first_arg.value` == `2` → `2.count('%s')` → 💥💥💥

Reply:
IMO, we should only support `%s` and `%(symbol)s`.  The other formatters will be confusing for translators.

Reply:
Counting `%(symbol)s` is useless, parenthesized placeholders **must** be named.

Let's at least count `%.2f`. I've got 12 hits for them in i18n string (could only search for those on a single line). At least one of those is broken as per this PR rationals:

```
~/Odoo/community/addons/hr_holidays/models/hr_leave.py:
  829                  short_leave_name = leave.name or time_off_type_display or _('Time Off')
  830                  if leave.leave_type_request_unit == 'hour':
  831:                     leave.display_name = _("%s: %.2f hours", short_leave_name, leave.number_of_hours_display)
  832                  else:
  833:                     leave.display_name = _("%s: %.2f days", short_leave_name, leave.number_of_days)
```

At the moment `_()` allows for all those complicated placeholders. I don't see the point of simplifying the linter if the gettext function itself is not simplified too.

Reply:
There are at least 75 `%d`, 14 of them seem broken.

There is also 1 broken `%g`

```
~/Odoo/community/addons/hr_holidays/models/hr_leave_type.py:
  297                  name = "{name} ({count})".format(
  298                      name=name,
  299:                     count=_('%g remaining out of %g') % (
  300                          float_round(record.virtual_remaining_leaves, precision_digits=2) or 0.0,
  301                          float_round(record.max_leaves, precision_digits=2) or 0.0,
```

Reply:
Discussed with rco, he agrees on this regexp linter